analyzer/runtime/extract.go

```diff
@@ -354,12 +354,16 @@ func ExtractRound(blockHeader block.Header, txrs []*sdkClient.TransactionWithRes
 			eventData.TxHash = blockTransactionData.Hash
 		}
 		if !res.FoundGasUsedEvent {
+			// Early versions of runtimes didn't emit a GasUsed event.
 			if (txr.Result.IsUnknown() || txr.Result.IsSuccess()) && tx != nil {
 				// Treat as if it used all the gas.
 				logger.Info("tx didn't emit a core.GasUsed event, assuming it used max allowed gas", "tx_hash", txr.Tx.Hash(), "assumed_gas_used", tx.AuthInfo.Fee.Gas)
 				res.TxGasUsed = tx.AuthInfo.Fee.Gas
-			} else { //nolint:staticcheck
+			} else {
 				// Inaccurate: Treat as not using any gas.
+				// TODO: Decode the tx; if it failed with "out of gas", assume it used all the gas.
```

Comment:
eh let's discuss that outside of this pr

Reply:
Possibly. Or we could do what I described in the comment, or something even more complex, like your suggestion from Slack: 
> in that case we can do
> auth failed -> no gas used
> insufficient balance (outside evm) -> no gas used
> most things within evm -> all gas used

Or we could leave things as-is, which I argued for in slack (but didn't update this TODO afterwards):

> I tried charging full gas_limit to all txs, and the fallout took me a while to figure out ...
>
> It turns out that some failed txs have a huge gas_limit; for example, I hit one with MAX_INT64. When you add other txs' gas limits to it, the block gas limit overflows int64 (and could easily overflow uint64). So that blows up in other places that expect total block gas to not exceed uint64; the db even uses int64. *[GitHub edit: Maybe we'd avoid this with the full three-pronged approach above, but I have no impulse to play around more; see also below.]*
> 
> Long story short, I'm leaving failed txs with no GasUsed event as-is, meaning we estimate them to have used 0 gas.
> I also like this product-wise. **If the gas we show is going to be a little wrong (and it always will be, no matter how we refine our heuristic), it should at least be easy to explain** on the rare occasion when we have to acknowledge the discrepancy.

So -- removing the TODO.